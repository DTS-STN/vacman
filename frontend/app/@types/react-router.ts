import 'react-router';

import type { Namespace } from 'i18next';

declare module 'react-router' {
  interface AppLoadContext {
    nonce: string;
    session: AppSession;
  }

  /**
   * Route handles should export an i18n namespace, if necessary.
   */
  interface RouteHandle {
    i18nNamespace?: Namespace;
  }

  /**
   * A route module exports an optional RouteHandle.
   */
  interface RouteModule {
    handle?: RouteHandle;
  }

  /**
   * Override the default React Router RouteModules
   * to include the new RouteModule type.
   */
  interface RouteModules extends Record<string, RouteModule | undefined> {}
}

export {};
