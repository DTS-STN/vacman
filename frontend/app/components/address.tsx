import type { ComponentProps, JSX } from 'react';

import { formatAddress } from '~/utils/string-utils';
import { cn } from '~/utils/tailwind-utils';

type AddressProps = ComponentProps<'address'> & {
  address: Parameters<typeof formatAddress>[0];
  format?: Parameters<typeof formatAddress>[1];
};

export function Address({ address, format, className, ...rest }: AddressProps): JSX.Element {
  return (
    <address className={cn('whitespace-pre-wrap not-italic', className)} data-testid="address-id" {...rest}>
      {formatAddress(address, format)}
    </address>
  );
}
