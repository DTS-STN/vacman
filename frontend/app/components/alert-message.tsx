import type { Ref } from 'react';

import type { IconDefinition } from '@fortawesome/free-solid-svg-icons';
import { faTriangleExclamation, faCircleExclamation, faCircleCheck } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface AlertMessageProps {
  message: string;
  type: 'success' | 'info' | 'error';
  ref?: Ref<HTMLDivElement>;
}

const styles: Record<NonNullable<AlertMessageProps['type']>, string> = {
  success: `border-green-600 bg-green-100`,
  error: `border-red-800 bg-red-100`,
  info: `border-sky-700 bg-sky-100`,
};

const icons: Record<NonNullable<AlertMessageProps['type']>, IconDefinition> = {
  success: faCircleCheck,
  error: faTriangleExclamation,
  info: faCircleExclamation,
};

export function AlertMessage({ ref, message, type = 'info' }: AlertMessageProps) {
  return (
    <div ref={ref} className={`${styles[type]} flex w-full items-center border-l-4 p-2`}>
      <FontAwesomeIcon
        icon={icons[type]}
        className={`${type === 'success' ? 'text-green-600' : type === 'error' ? 'text-red-800' : 'text-sky-700'} mr-2`}
      />
      {message}
    </div>
  );
}
