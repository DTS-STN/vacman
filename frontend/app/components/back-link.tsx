import type { Params, Path } from 'react-router';

import { faAngleLeft } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useTranslation } from 'react-i18next';

import { InlineLink } from '~/components/links';
import type { I18nRouteFile } from '~/i18n-routes';
import { cn } from '~/utils/tailwind-utils';

interface BackLinkProps {
  file: I18nRouteFile;
  params?: Params;
  className?: string;
  hash?: Path['hash'];
  search?: Path['search'];
  translationKey?: string;
}

export function BackLink({ className, file, params, hash, search, translationKey = 'app:profile.back' }: BackLinkProps) {
  const { t } = useTranslation();
  const linkText = t(translationKey);
  return (
    <InlineLink
      className={cn('inline-flex items-center', className)}
      file={file}
      params={params}
      hash={hash}
      search={search}
      aria-label={linkText}
    >
      <FontAwesomeIcon icon={faAngleLeft} />
      {linkText}
    </InlineLink>
  );
}
