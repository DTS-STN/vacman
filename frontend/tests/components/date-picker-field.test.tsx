import { createRoutesStub } from 'react-router';

import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { DatePickerField } from '~/components/date-picker-field';

describe('DatePickerField', () => {
  it('should render date picker field component', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/',
        Component: () => (
          <DatePickerField
            id="id"
            legend="legend date picker field test"
            names={{
              year: 'yearTest',
              month: 'monthTest',
              day: 'dayTest',
            }}
          />
        ),
      },
    ]);

    const { container } = render(<RoutesStub />);

    expect(container).toMatchSnapshot('expected html');
  });

  it('should render date picker field component with required', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/',
        Component: () => (
          <DatePickerField
            id="id"
            legend="legend date picker field test"
            names={{
              year: 'yearTest',
              month: 'monthTest',
              day: 'dayTest',
            }}
            required
          />
        ),
      },
    ]);

    const { container } = render(<RoutesStub />);

    expect(container).toMatchSnapshot('expected html');
  });

  it('should render date picker field component with help message primary and help message secondary', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/',
        Component: () => (
          <DatePickerField
            helpMessagePrimary="help message primary"
            helpMessageSecondary="help message secondary"
            id="id"
            legend="legend date picker field test"
            names={{
              year: 'yearTest',
              month: 'monthTest',
              day: 'dayTest',
            }}
          />
        ),
      },
    ]);

    const { container } = render(<RoutesStub />);

    expect(container).toMatchSnapshot('expected html');
  });

  it('should render date picker field component with default value as string', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/',
        Component: () => (
          <DatePickerField
            defaultValue={'2025-01-01'}
            id="id"
            legend="legend date picker field test"
            names={{
              year: 'yearTest',
              month: 'monthTest',
              day: 'dayTest',
            }}
          />
        ),
      },
    ]);

    const { container } = render(<RoutesStub />);

    expect(container).toMatchSnapshot('expected html');
  });

  it('should render date picker field component with default value as object', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/',
        Component: () => (
          <DatePickerField
            defaultValue={{
              year: '2025',
              month: '01',
              day: '31',
            }}
            id="id"
            legend="legend date picker field test"
            names={{
              year: 'yearTest',
              month: 'monthTest',
              day: 'dayTest',
            }}
          />
        ),
      },
    ]);

    const { container } = render(<RoutesStub />);

    expect(container).toMatchSnapshot('expected html');
  });

  it('should render date picker field component with error message', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/',
        Component: () => (
          <DatePickerField
            id="id"
            legend="legend date picker field test"
            names={{
              year: 'yearTest',
              month: 'monthTest',
              day: 'dayTest',
            }}
            errorMessages={{
              all: 'test error all',
              day: 'test error day',
              month: 'test error month',
              year: 'test error year',
            }}
          />
        ),
      },
    ]);

    const { container } = render(<RoutesStub />);

    expect(container).toMatchSnapshot('expected html');
  });
});
