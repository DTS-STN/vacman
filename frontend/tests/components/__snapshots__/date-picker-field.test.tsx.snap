// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DatePickerField > should render date picker field component > expected html 1`] = `
<div>
  <div
    id="date-picker-id"
  >
    <fieldset
      class="space-y-2"
    >
      <legend
        class="block"
        id="date-picker-id-legend"
      >
        <span
          class="font-semibold"
        >
          legend date picker field test
        </span>
      </legend>
      <div
        class="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2"
      >
        <div
          class="space-y-2"
          id="date-picker-id-month"
        >
          <label
            class="block"
            for="date-picker-id-month-select"
            id="date-picker-id-month-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.month.label
            </span>
          </label>
          <select
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-month-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-auto"
            id="date-picker-id-month-select"
            name="monthTest"
          >
            <option
              disabled=""
              hidden=""
              id="date-picker-id-month-option-unselected"
              selected=""
              value=""
            >
              gcweb:date-picker.month.placeholder
            </option>
            <option
              id="date-picker-id-month-option-1"
              value="01"
            >
              January
            </option>
            <option
              id="date-picker-id-month-option-2"
              value="02"
            >
              February
            </option>
            <option
              id="date-picker-id-month-option-3"
              value="03"
            >
              March
            </option>
            <option
              id="date-picker-id-month-option-4"
              value="04"
            >
              April
            </option>
            <option
              id="date-picker-id-month-option-5"
              value="05"
            >
              May
            </option>
            <option
              id="date-picker-id-month-option-6"
              value="06"
            >
              June
            </option>
            <option
              id="date-picker-id-month-option-7"
              value="07"
            >
              July
            </option>
            <option
              id="date-picker-id-month-option-8"
              value="08"
            >
              August
            </option>
            <option
              id="date-picker-id-month-option-9"
              value="09"
            >
              September
            </option>
            <option
              id="date-picker-id-month-option-10"
              value="10"
            >
              October
            </option>
            <option
              id="date-picker-id-month-option-11"
              value="11"
            >
              November
            </option>
            <option
              id="date-picker-id-month-option-12"
              value="12"
            >
              December
            </option>
          </select>
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-day"
        >
          <label
            class="block"
            for="date-picker-id-day-input"
            id="date-picker-id-day-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.day.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-day-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-20"
            id="date-picker-id-day-input"
            inputmode="numeric"
            max="31"
            maxlength="2"
            min="1"
            name="dayTest"
            pattern="[0-9]*"
            type="text"
          />
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-year"
        >
          <label
            class="block"
            for="date-picker-id-year-input"
            id="date-picker-id-year-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.year.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-year-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-32"
            id="date-picker-id-year-input"
            inputmode="numeric"
            maxlength="4"
            min="1900"
            name="yearTest"
            pattern="[0-9]*"
            type="text"
          />
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`DatePickerField > should render date picker field component with default value as object > expected html 1`] = `
<div>
  <div
    id="date-picker-id"
  >
    <fieldset
      class="space-y-2"
    >
      <legend
        class="block"
        id="date-picker-id-legend"
      >
        <span
          class="font-semibold"
        >
          legend date picker field test
        </span>
      </legend>
      <div
        class="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2"
      >
        <div
          class="space-y-2"
          id="date-picker-id-month"
        >
          <label
            class="block"
            for="date-picker-id-month-select"
            id="date-picker-id-month-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.month.label
            </span>
          </label>
          <select
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-month-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-auto"
            id="date-picker-id-month-select"
            name="monthTest"
          >
            <option
              disabled=""
              hidden=""
              id="date-picker-id-month-option-unselected"
              value=""
            >
              gcweb:date-picker.month.placeholder
            </option>
            <option
              id="date-picker-id-month-option-1"
              selected=""
              value="01"
            >
              January
            </option>
            <option
              id="date-picker-id-month-option-2"
              value="02"
            >
              February
            </option>
            <option
              id="date-picker-id-month-option-3"
              value="03"
            >
              March
            </option>
            <option
              id="date-picker-id-month-option-4"
              value="04"
            >
              April
            </option>
            <option
              id="date-picker-id-month-option-5"
              value="05"
            >
              May
            </option>
            <option
              id="date-picker-id-month-option-6"
              value="06"
            >
              June
            </option>
            <option
              id="date-picker-id-month-option-7"
              value="07"
            >
              July
            </option>
            <option
              id="date-picker-id-month-option-8"
              value="08"
            >
              August
            </option>
            <option
              id="date-picker-id-month-option-9"
              value="09"
            >
              September
            </option>
            <option
              id="date-picker-id-month-option-10"
              value="10"
            >
              October
            </option>
            <option
              id="date-picker-id-month-option-11"
              value="11"
            >
              November
            </option>
            <option
              id="date-picker-id-month-option-12"
              value="12"
            >
              December
            </option>
          </select>
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-day"
        >
          <label
            class="block"
            for="date-picker-id-day-input"
            id="date-picker-id-day-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.day.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-day-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-20"
            id="date-picker-id-day-input"
            inputmode="numeric"
            max="31"
            maxlength="2"
            min="1"
            name="dayTest"
            pattern="[0-9]*"
            type="text"
            value="31"
          />
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-year"
        >
          <label
            class="block"
            for="date-picker-id-year-input"
            id="date-picker-id-year-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.year.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-year-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-32"
            id="date-picker-id-year-input"
            inputmode="numeric"
            maxlength="4"
            min="1900"
            name="yearTest"
            pattern="[0-9]*"
            type="text"
            value="2025"
          />
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`DatePickerField > should render date picker field component with default value as string > expected html 1`] = `
<div>
  <div
    id="date-picker-id"
  >
    <fieldset
      class="space-y-2"
    >
      <legend
        class="block"
        id="date-picker-id-legend"
      >
        <span
          class="font-semibold"
        >
          legend date picker field test
        </span>
      </legend>
      <div
        class="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2"
      >
        <div
          class="space-y-2"
          id="date-picker-id-month"
        >
          <label
            class="block"
            for="date-picker-id-month-select"
            id="date-picker-id-month-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.month.label
            </span>
          </label>
          <select
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-month-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-auto"
            id="date-picker-id-month-select"
            name="monthTest"
          >
            <option
              disabled=""
              hidden=""
              id="date-picker-id-month-option-unselected"
              value=""
            >
              gcweb:date-picker.month.placeholder
            </option>
            <option
              id="date-picker-id-month-option-1"
              selected=""
              value="01"
            >
              January
            </option>
            <option
              id="date-picker-id-month-option-2"
              value="02"
            >
              February
            </option>
            <option
              id="date-picker-id-month-option-3"
              value="03"
            >
              March
            </option>
            <option
              id="date-picker-id-month-option-4"
              value="04"
            >
              April
            </option>
            <option
              id="date-picker-id-month-option-5"
              value="05"
            >
              May
            </option>
            <option
              id="date-picker-id-month-option-6"
              value="06"
            >
              June
            </option>
            <option
              id="date-picker-id-month-option-7"
              value="07"
            >
              July
            </option>
            <option
              id="date-picker-id-month-option-8"
              value="08"
            >
              August
            </option>
            <option
              id="date-picker-id-month-option-9"
              value="09"
            >
              September
            </option>
            <option
              id="date-picker-id-month-option-10"
              value="10"
            >
              October
            </option>
            <option
              id="date-picker-id-month-option-11"
              value="11"
            >
              November
            </option>
            <option
              id="date-picker-id-month-option-12"
              value="12"
            >
              December
            </option>
          </select>
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-day"
        >
          <label
            class="block"
            for="date-picker-id-day-input"
            id="date-picker-id-day-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.day.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-day-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-20"
            id="date-picker-id-day-input"
            inputmode="numeric"
            max="31"
            maxlength="2"
            min="1"
            name="dayTest"
            pattern="[0-9]*"
            type="text"
            value="01"
          />
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-year"
        >
          <label
            class="block"
            for="date-picker-id-year-input"
            id="date-picker-id-year-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.year.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-year-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-32"
            id="date-picker-id-year-input"
            inputmode="numeric"
            maxlength="4"
            min="1900"
            name="yearTest"
            pattern="[0-9]*"
            type="text"
            value="2025"
          />
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`DatePickerField > should render date picker field component with error message > expected html 1`] = `
<div>
  <div
    id="date-picker-id"
  >
    <fieldset
      class="space-y-2"
    >
      <legend
        class="block"
        id="date-picker-id-legend"
      >
        <span
          class="font-semibold"
        >
          legend date picker field test
        </span>
      </legend>
      <div
        class="space-y-2"
      >
        <span
          class="inline-block max-w-prose border-l-2 border-red-600 bg-red-50 px-3 py-1"
          data-testid="input-error-test-id"
          id="date-picker-id-error-all"
          role="alert"
        >
          test error all
        </span>
        <p>
          <span
            class="inline-block max-w-prose border-l-2 border-red-600 bg-red-50 px-3 py-1"
            data-testid="input-error-test-id"
            id="date-picker-id-error-month"
            role="alert"
          >
            test error month
          </span>
        </p>
        <p>
          <span
            class="inline-block max-w-prose border-l-2 border-red-600 bg-red-50 px-3 py-1"
            data-testid="input-error-test-id"
            id="date-picker-id-error-day"
            role="alert"
          >
            test error day
          </span>
        </p>
        <p>
          <span
            class="inline-block max-w-prose border-l-2 border-red-600 bg-red-50 px-3 py-1"
            data-testid="input-error-test-id"
            id="date-picker-id-error-year"
            role="alert"
          >
            test error year
          </span>
        </p>
      </div>
      <div
        class="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2"
      >
        <div
          class="space-y-2"
          id="date-picker-id-month"
        >
          <label
            class="block"
            for="date-picker-id-month-select"
            id="date-picker-id-month-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.month.label
            </span>
          </label>
          <select
            aria-describedby="date-picker-id-legend"
            aria-errormessage="date-picker-id-error-all date-picker-id-error-month"
            aria-invalid="true"
            aria-labelledby="date-picker-id-month-label"
            class="rounded-lg block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 border-red-500 focus:border-red-500 focus:ring-red-500 w-full sm:w-auto"
            id="date-picker-id-month-select"
            name="monthTest"
          >
            <option
              disabled=""
              hidden=""
              id="date-picker-id-month-option-unselected"
              selected=""
              value=""
            >
              gcweb:date-picker.month.placeholder
            </option>
            <option
              id="date-picker-id-month-option-1"
              value="01"
            >
              January
            </option>
            <option
              id="date-picker-id-month-option-2"
              value="02"
            >
              February
            </option>
            <option
              id="date-picker-id-month-option-3"
              value="03"
            >
              March
            </option>
            <option
              id="date-picker-id-month-option-4"
              value="04"
            >
              April
            </option>
            <option
              id="date-picker-id-month-option-5"
              value="05"
            >
              May
            </option>
            <option
              id="date-picker-id-month-option-6"
              value="06"
            >
              June
            </option>
            <option
              id="date-picker-id-month-option-7"
              value="07"
            >
              July
            </option>
            <option
              id="date-picker-id-month-option-8"
              value="08"
            >
              August
            </option>
            <option
              id="date-picker-id-month-option-9"
              value="09"
            >
              September
            </option>
            <option
              id="date-picker-id-month-option-10"
              value="10"
            >
              October
            </option>
            <option
              id="date-picker-id-month-option-11"
              value="11"
            >
              November
            </option>
            <option
              id="date-picker-id-month-option-12"
              value="12"
            >
              December
            </option>
          </select>
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-day"
        >
          <label
            class="block"
            for="date-picker-id-day-input"
            id="date-picker-id-day-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.day.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-errormessage="date-picker-id-error-all date-picker-id-error-day"
            aria-invalid="true"
            aria-labelledby="date-picker-id-day-label"
            class="rounded-lg block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 border-red-500 focus:border-red-500 focus:ring-red-500 w-full sm:w-20"
            id="date-picker-id-day-input"
            inputmode="numeric"
            max="31"
            maxlength="2"
            min="1"
            name="dayTest"
            pattern="[0-9]*"
            type="text"
          />
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-year"
        >
          <label
            class="block"
            for="date-picker-id-year-input"
            id="date-picker-id-year-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.year.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-errormessage="date-picker-id-error-all date-picker-id-error-year"
            aria-invalid="true"
            aria-labelledby="date-picker-id-year-label"
            class="rounded-lg block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 border-red-500 focus:border-red-500 focus:ring-red-500 w-full sm:w-32"
            id="date-picker-id-year-input"
            inputmode="numeric"
            maxlength="4"
            min="1900"
            name="yearTest"
            pattern="[0-9]*"
            type="text"
          />
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`DatePickerField > should render date picker field component with help message primary and help message secondary > expected html 1`] = `
<div>
  <div
    id="date-picker-id"
  >
    <fieldset
      class="space-y-2"
    >
      <legend
        class="block"
        id="date-picker-id-legend"
      >
        <span
          class="font-semibold"
        >
          legend date picker field test
        </span>
      </legend>
      <span
        class="block max-w-prose text-gray-500"
        data-testid="input-help"
        id="date-picker-id-help-primary"
      >
        help message primary
      </span>
      <div
        class="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2"
      >
        <div
          class="space-y-2"
          id="date-picker-id-month"
        >
          <label
            class="block"
            for="date-picker-id-month-select"
            id="date-picker-id-month-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.month.label
            </span>
          </label>
          <select
            aria-describedby="date-picker-id-legend date-picker-id-help-primary date-picker-id-help-secondary"
            aria-invalid="false"
            aria-labelledby="date-picker-id-month-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-auto"
            id="date-picker-id-month-select"
            name="monthTest"
          >
            <option
              disabled=""
              hidden=""
              id="date-picker-id-month-option-unselected"
              selected=""
              value=""
            >
              gcweb:date-picker.month.placeholder
            </option>
            <option
              id="date-picker-id-month-option-1"
              value="01"
            >
              January
            </option>
            <option
              id="date-picker-id-month-option-2"
              value="02"
            >
              February
            </option>
            <option
              id="date-picker-id-month-option-3"
              value="03"
            >
              March
            </option>
            <option
              id="date-picker-id-month-option-4"
              value="04"
            >
              April
            </option>
            <option
              id="date-picker-id-month-option-5"
              value="05"
            >
              May
            </option>
            <option
              id="date-picker-id-month-option-6"
              value="06"
            >
              June
            </option>
            <option
              id="date-picker-id-month-option-7"
              value="07"
            >
              July
            </option>
            <option
              id="date-picker-id-month-option-8"
              value="08"
            >
              August
            </option>
            <option
              id="date-picker-id-month-option-9"
              value="09"
            >
              September
            </option>
            <option
              id="date-picker-id-month-option-10"
              value="10"
            >
              October
            </option>
            <option
              id="date-picker-id-month-option-11"
              value="11"
            >
              November
            </option>
            <option
              id="date-picker-id-month-option-12"
              value="12"
            >
              December
            </option>
          </select>
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-day"
        >
          <label
            class="block"
            for="date-picker-id-day-input"
            id="date-picker-id-day-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.day.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend date-picker-id-help-primary date-picker-id-help-secondary"
            aria-invalid="false"
            aria-labelledby="date-picker-id-day-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-20"
            id="date-picker-id-day-input"
            inputmode="numeric"
            max="31"
            maxlength="2"
            min="1"
            name="dayTest"
            pattern="[0-9]*"
            type="text"
          />
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-year"
        >
          <label
            class="block"
            for="date-picker-id-year-input"
            id="date-picker-id-year-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.year.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend date-picker-id-help-primary date-picker-id-help-secondary"
            aria-invalid="false"
            aria-labelledby="date-picker-id-year-label"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-32"
            id="date-picker-id-year-input"
            inputmode="numeric"
            maxlength="4"
            min="1900"
            name="yearTest"
            pattern="[0-9]*"
            type="text"
          />
        </div>
      </div>
      <span
        class="block max-w-prose text-gray-500"
        data-testid="input-help"
        id="date-picker-id-help-secondary"
      >
        help message secondary
      </span>
    </fieldset>
  </div>
</div>
`;

exports[`DatePickerField > should render date picker field component with required > expected html 1`] = `
<div>
  <div
    id="date-picker-id"
  >
    <fieldset
      class="space-y-2"
    >
      <legend
        class="block"
        id="date-picker-id-legend"
      >
        <span
          class="font-semibold"
        >
          legend date picker field test
        </span>
         
        <span
          aria-hidden="true"
        >
          (
          input.required
          )
        </span>
      </legend>
      <div
        class="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2"
      >
        <div
          class="space-y-2"
          id="date-picker-id-month"
        >
          <label
            class="block"
            for="date-picker-id-month-select"
            id="date-picker-id-month-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.month.label
            </span>
          </label>
          <select
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-month-label"
            aria-required="true"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-auto"
            id="date-picker-id-month-select"
            name="monthTest"
            required=""
          >
            <option
              disabled=""
              hidden=""
              id="date-picker-id-month-option-unselected"
              selected=""
              value=""
            >
              gcweb:date-picker.month.placeholder
            </option>
            <option
              id="date-picker-id-month-option-1"
              value="01"
            >
              January
            </option>
            <option
              id="date-picker-id-month-option-2"
              value="02"
            >
              February
            </option>
            <option
              id="date-picker-id-month-option-3"
              value="03"
            >
              March
            </option>
            <option
              id="date-picker-id-month-option-4"
              value="04"
            >
              April
            </option>
            <option
              id="date-picker-id-month-option-5"
              value="05"
            >
              May
            </option>
            <option
              id="date-picker-id-month-option-6"
              value="06"
            >
              June
            </option>
            <option
              id="date-picker-id-month-option-7"
              value="07"
            >
              July
            </option>
            <option
              id="date-picker-id-month-option-8"
              value="08"
            >
              August
            </option>
            <option
              id="date-picker-id-month-option-9"
              value="09"
            >
              September
            </option>
            <option
              id="date-picker-id-month-option-10"
              value="10"
            >
              October
            </option>
            <option
              id="date-picker-id-month-option-11"
              value="11"
            >
              November
            </option>
            <option
              id="date-picker-id-month-option-12"
              value="12"
            >
              December
            </option>
          </select>
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-day"
        >
          <label
            class="block"
            for="date-picker-id-day-input"
            id="date-picker-id-day-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.day.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-day-label"
            aria-required="true"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-20"
            id="date-picker-id-day-input"
            inputmode="numeric"
            max="31"
            maxlength="2"
            min="1"
            name="dayTest"
            pattern="[0-9]*"
            required=""
            type="text"
          />
        </div>
        <div
          class="space-y-2"
          id="date-picker-id-year"
        >
          <label
            class="block"
            for="date-picker-id-year-input"
            id="date-picker-id-year-label"
          >
            <span
              class="font-semibold"
            >
              gcweb:date-picker.year.label
            </span>
          </label>
          <input
            aria-describedby="date-picker-id-legend"
            aria-invalid="false"
            aria-labelledby="date-picker-id-year-label"
            aria-required="true"
            class="rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 w-full sm:w-32"
            id="date-picker-id-year-input"
            inputmode="numeric"
            maxlength="4"
            min="1900"
            name="yearTest"
            pattern="[0-9]*"
            required=""
            type="text"
          />
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;
